ios/brave-ios/Sources/DataImporter/UI/DataImporterView.swift

```diff
@@ -0,0 +1,361 @@
+  private var dismiss
+
+  @State
+  private var showSuccessView = false
+
+  @State
+  private var showFailureView = false
+
+  @State
+  private var showLoadingView = false
+
+  @State
+  private var showMainView = true
+
+  @State
+  private var importerInfo = ImporterInfo(shouldShowFileImporter: false)
+
+  @ObservedObject
+  private var model = DataImportModel()
+
+  private var openURL: (URL) -> Void
+  private var onDismiss: () -> Void
+
+  public init(openURL: @escaping (URL) -> Void, onDismiss: @escaping () -> Void) {
+    self.openURL = openURL
+    self.onDismiss = onDismiss
+  }
+
+  public var body: some View {
+    ZStack {
+      if model.importState == .success {
+        DataImporterStateView(kind: .success, model: model) {
+          model.removeZipFile()
+
+          onDismiss()
+          dismiss()
+        }
+        .padding(16.0)
+        .frame(maxWidth: .infinity, maxHeight: .infinity)
+        .background(Color(braveSystemName: .pageBackground))
+        .toolbar(.hidden, for: .navigationBar)
+        .opacity(showSuccessView ? 1.0 : 0.0)
+        .onAppear {
+          withAnimation(.easeInOut(duration: 0.5)) {
+            showSuccessView = true
+          }
+        }
+        .onDisappear {
+          withAnimation(.easeInOut(duration: 0.5)) {
+            showSuccessView = false
+          }
+        }
+      } else if model.importState == .failure {
+        DataImporterStateView(kind: .failure, model: model) {
+          model.removeZipFile()
+          model.resetAllStates()
+        }
+        .padding(16.0)
+        .frame(maxWidth: .infinity, maxHeight: .infinity)
+        .background(Color(braveSystemName: .pageBackground))
+        .toolbar(.hidden, for: .navigationBar)
+        .opacity(showFailureView ? 1.0 : 0.0)
+        .onAppear {
+          withAnimation(.easeInOut(duration: 0.5)) {
+            showFailureView = true
+          }
+        }
+        .onDisappear {
+          withAnimation(.easeInOut(duration: 0.5)) {
+            showFailureView = false
+          }
+        }
+      } else if model.importState == .importing {
+        DataImporterLoadingView()
+          .padding(16.0)
+          .frame(maxWidth: .infinity, maxHeight: .infinity)
+          .background(Color(braveSystemName: .pageBackground))
+          .toolbar(.hidden, for: .navigationBar)
+          .opacity(showLoadingView ? 1.0 : 0.0)
+          .onAppear {
+            withAnimation(.easeInOut(duration: 0.5)) {
+              showLoadingView = true
+            }
+          }
+          .onDisappear {
+            withAnimation(.easeInOut(duration: 0.5)) {
+              showLoadingView = false
+            }
+          }
+      } else {
+        GeometryReader { proxy in
+          ScrollView {
+            mainView()
+              .frame(minHeight: proxy.size.height)
+          }
+          .background(Color(braveSystemName: .pageBackground))
+        }
+        .sheet(
+          isPresented: $model.isLoadingProfiles,
```

Comment:
These sheets are explicitly associated with the main view only it seems? Is that correct? If for instance the import state is any causing any of the other views to appear the sheets will be dismissed, should these go on the container view instead?